File: source-code/Server/backend/src/main/java/com/iasdietideals24/backend/mapstruct/dto/AstaSilenziosaDto.java
package com.iasdietideals24.backend.mapstruct.dto;

import com.iasdietideals24.backend.mapstruct.dto.shallows.OffertaShallowDto;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Set;

@Getter
@Setter
@NoArgsConstructor
public class AstaSilenziosaDto extends AstaDiVenditoreDto {

    private Set<OffertaShallowDto> offerteRicevuteShallow;
}

File: source-code/Client/frontendandroid/app/src/main/java/com/iasdietideals24/dietideals24/utilities/dto/AstaSilenziosaDto.kt
package com.iasdietideals24.dietideals24.utilities.dto

import com.iasdietideals24.dietideals24.utilities.dto.shallows.AccountShallowDto
import com.iasdietideals24.dietideals24.utilities.dto.shallows.NotificaShallowDto
import com.iasdietideals24.dietideals24.utilities.dto.shallows.OffertaShallowDto
import lombok.Getter
import lombok.NoArgsConstructor
import lombok.Setter
import java.time.LocalDate
import java.time.LocalTime

@Getter
@Setter
@NoArgsConstructor
class AstaSilenziosaDto {
    private val idAsta: Long? = null

    private val categoria: String? = null

    private val nome: String? = null

    private val descrizione: String? = null

    private val dataScadenza: LocalDate? = null

    private val oraScadenza: LocalTime? = null

    private val immagine: ByteArray? = null

    private val notificheAssociateShallow: Set<NotificaShallowDto>? = null

    private val proprietarioShallow: AccountShallowDto? = null

    private val offerteRicevuteShallow: Set<OffertaShallowDto>? = null
}
